packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,585 @@
+}
+
+enum BarcodePhoneType {
+  /// Unknown phone type.
+  Unknown,
+
+  /// Barcode work phone type.
+  Work,
+
+  /// Barcode home phone type.
+  Home,
+
+  /// Barcode fax phone type.
+  Fax,
+
+  /// Barcode mobile phone type.
+  Mobile,
+}
+
+/// An sms message from an 'SMS:' or similar QRCode type.
+class BarcodeSMS {
+  BarcodeSMS._(Map<dynamic, dynamic> data)
+      : message = data['message'],
+        phoneNumber = data['phone_number'];
+
+  final String message;
+  final String phoneNumber;
+}
+
+/// A URL and title from a 'MEBKM:' or similar QRCode type.
+class BarcodeURLBookmark {
+  BarcodeURLBookmark._(Map<dynamic, dynamic> data)
+      : title = data['title'],
+        url = data['url'];
+
+  final String title;
+  final String url;
+}
+
+/// A wifi network parameters from a 'WIFI:' or similar QRCode type.
+class BarcodeWiFi {
+  BarcodeWiFi._(Map<dynamic, dynamic> data)
+      : ssid = data['ssid'],
+        password = data['password'],
+        encryptionType =
+            BarcodeWiFiEncryptionType.values.elementAt(data['encryption_type']);
+
+  final String ssid;
+  final String password;
+
+  /// Gets the encryption type of the WIFI
+  ///
+  /// See all [BarcodeWiFiEncryptionType]
+  final BarcodeWiFiEncryptionType encryptionType;
+}
+
+/// Wifi encryption type constants.
+enum BarcodeWiFiEncryptionType {
+  /// Barcode unknown Wi-Fi encryption type.
+  Unknown,
+
+  /// Barcode open Wi-Fi encryption type.
+  Open,
+
+  /// Barcode WPA Wi-Fi encryption type.
+  WPA,
+
+  /// Barcode WEP Wi-Fi encryption type.
+  WEP,
+}
+
+/// GPS coordinates from a 'GEO:' or similar QRCode type.
+class BarcodeGeoPoint {
+  BarcodeGeoPoint._(Map<dynamic, dynamic> data)
+      : latitude = data['latitude'],
+        longitude = data['longitude'];
+  final double latitude;
+  final double longitude;
+}
+
+/// A person's or organization's business card.
+class BarcodeContactInfo {
+  BarcodeContactInfo._(Map<dynamic, dynamic> data)
+      : addresses = data['addresses'] == null
+            ? null
+            : data['addresses']
+                .map<BarcodeAddress>((dynamic item) => BarcodeAddress._(item))
+                .toList(),
+        emails = data['emails'] == null
+            ? null
+            : data['emails']
+                .map<BarcodeEmail>((dynamic item) => BarcodeEmail._(item))
+                .toList(),
+        name = data['name'] == null ? null : BarcodePersonName._(data['name']),
+        phones = data['phones'] == null
+            ? null
+            : data['phones']
+                .map<BarcodePhone>((dynamic item) => BarcodePhone._(item))
+                .toList(),
+        urls = data['urls'] == null
+            ? null
+            : data['urls'].map<String>((dynamic item) {
+                final String s = item;
+                return s;
+              }).toList(),
+        jobTitle = data['job_title'],
+        organization = data['organization'];
+
+  /// Gets contact person's addresses.
```

Comment:
You use `Gets` for the variables in this class and all the classes below this one.